include/AST/Types/FunctionTy.hpp

```diff
@@ -0,0 +1,48 @@
+#ifndef GLU_AST_TYPES_FUNCTIONTY_HPP
+#define GLU_AST_TYPES_FUNCTIONTY_HPP
+
+#include "TypeBase.hpp"
+#include <vector>
+
+namespace glu::types {
+
+/// @brief FunctionTy is a class that represents a function type in the AST.
+class FunctionTy : public TypeBase {
+private:
+    std::vector<TypeBase *> const _parameters;
+    TypeBase *_returnType;
+
+public:
+    /// @brief Constructor for the FunctionTy class.
+    /// @param parameters A vector of TypeBase pointers representing the
+    /// parameters of the function.
+    /// @param returnType A TypeBase pointer representing the return type of the
+    /// function.
+    FunctionTy(std::vector<TypeBase *> const &&parameters, TypeBase *returnType)
```

Comment:
what is the point of a const rvalue reference??
better take an owned value, but rvalue-ref works too